### PR TITLE
Fetch presigned URLs dynamically

### PR DESCRIPTION
**Pull request by sam-at-luther**

### CODE REVIEW

File: common.fabric.mk [hunk 1]
```diff
@@ -361,5 +361,9 @@ compile-phylum-%: $$(shell find -L ../$$* -name "*.lisp" 2>/dev/null | grep -Fv
 	cp $^    ./build/phylum_$*/src/
 	cd       ./build/phylum_$*/src && ls && rm -f ./../phylum.zip && zip ./../phylum.zip $(notdir $^)
 
-${CC_PATH}:
+${CC_PATH}: ${PRESIGNED_PATH}
 	${PROJECT_REL_DIR}/scripts/obtain-cc.sh
+	touch $@
```
Comment (bmatsuo-at-luthersystems) [inline common.fabric.mk:366]: Do you really want to `touch $@` here? CC_PATH isn't a "dummy" path is it? It's the actual path of the cc tarball. right?

File: common.mk [hunk 2]
```diff
@@ -89,3 +94,19 @@ echo\:%:
 docker-pull/%: id=$(shell docker image inspect -f "{{.Id}}" $* 2>/dev/null)
 docker-pull/%:
 	@[[ -n "${id}" ]] || { echo "retrieving $*" && docker pull $*; }
+
+${STATIC_PRESIGN_DUMMY}: ${LICENSE_FILE}
+	${MKDIR_P} $(dir $@)
+	./scripts/obtain-presigned.sh
+	touch $@
```
Comment (bmatsuo-at-luthersystems) [inline common.mk:101]: Same question.

File: common.mk [hunk 3]
```diff
@@ -89,3 +94,19 @@ echo\:%:
...
+
+${PRESIGNED_PATH}: ${STATIC_PRESIGN_DUMMY}
+	@
+
+${STATIC_PLUGINS_DUMMY}: ${PRESIGNED_PATH}
+	${MKDIR_P} $(dir $@)
+	./scripts/obtain-plugin.sh
+	touch $@
```
Comment (bmatsuo-at-luthersystems) [inline common.mk:109]: Same question.

**Review by bmatsuo-at-luthersystems — APPROVED** [review]

File: common.mk (hunk 2 above)
Comment (bmatsuo-at-luthersystems) [reply common.mk:101]: Well I guess these are dummies. Should they actually be dummies? What happens when the presigned url expires?

File: common.fabric.mk (hunk 1 above)
Comment (sam-at-luther) [reply common.fabric.mk:366]: So apparently the mod time from the file I was downloading from s3 was interfering with `make`, and this fixed it. Have you seen that before?

File: common.mk (hunk 3 above)
Comment (sam-at-luther) [reply common.mk:109]: this is a DUMMY file, since the script duownloads multiple artifacts.

Comment (sam-at-luther) [reply common.mk:109]: this is a DUMMY file, since the script duownloads multiple artifacts.

File: common.mk (hunk 2 above)
Comment (sam-at-luther) [reply common.mk:101]: We want to re-download the presigned file if the license file changes or SUBSTRATE_VERSION changes. The presigned links are only for an hour so the presumption is that they run `make up` which generates the presigned links and immediately downloads the artifacts.

File: common.fabric.mk (hunk 1 above)
Comment (bmatsuo-at-luthersystems) [reply common.fabric.mk:366]: Oh. Huh. I have never noticed that.. But I guess the s3 cli might be trying to set the mod time to be helpful?

File: common.mk (hunk 2 above)
Comment (bmatsuo-at-luthersystems) [reply common.mk:101]: I see now. That should work fine

**Review by bmatsuo-at-luthersystems — APPROVED** [review]

**Pull request merged**